### PR TITLE
SDI-33 Divide batch into page chunks

### PR DESCRIPTION
**Pull request by andymarke**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/data/MigrationContext.kt [hunk 1]
```diff
@@ -3,6 +3,8 @@ package uk.gov.justice.digital.hmpps.prisonerfromnomismigration.data
 import java.time.LocalDateTime
 import java.time.format.DateTimeFormatter
 
-class MigrationContext<T>(val migrationId: String, val estimatedCount: Long, val filter: T)
+class MigrationContext<T>(val migrationId: String, val estimatedCount: Long, val body: T) {
```
Comment (andymarke) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/data/MigrationContext.kt:6]: renamed to body, more flexible then "filter"

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/listeners/MigrationMessageListener.kt [hunk 2]
```diff
@@ -28,6 +29,7 @@ class MigrationMessageListener(
     val migrationMessage: MigrationMessage<*> = message.fromJson()
     when (migrationMessage.type) {
       MIGRATE_VISITS -> visitsMigrationService.migrateVisitsByPage(context(message.fromJson()))
+      MIGRATE_VISITS_BY_PAGE -> visitsMigrationService.migrateVisitsForPage(context(message.fromJson()))
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/listeners/MigrationMessageListener.kt:32]: this reads a bit odd `MIGRATE_VISITS` calls `migrateVisitsByPage` but `MIGRATE_VISITS_BY_PAGE` calls something else

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/listeners/MigrationMessageListener.kt (hunk 2 above)
Comment (andymarke) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/listeners/MigrationMessageListener.kt:32]: Yes, they are sort of state transitions I suppose, 
MIGRATE_VISITS -> MIGRATE_VISITS_BY_PAGE and each function is the transition to the next stage, hence it looks odd

**Pull request merged**